avocado/utils/cpu.py

```diff
@@ -188,3 +189,56 @@ def offline(cpu):
     if _get_cpu_status(cpu):
         return 1
     return 0
+
+
+def get_cpu_idle():
+    """
+    Get current cpu idle values
+    """
+    cpus_list = cpu_online_list()
+    states = range(len(glob.glob("/sys/devices/system/cpu/cpu0/cpuidle/state*")))
+    cpu_idle_state = [[0]*len(states)]*len(cpus_list)
+    for c in cpus_list:
+        for state in states:
+            state_file = "/sys/devices/system/cpu/cpu%s/cpuidle/state%s/disable" % (c, state)
+            try:
+                cpu_idle_state[c][state] = int(open(state_file).read())
+            except Exception:
+                pass  # Ignore if states are not available
+    return cpu_idle_state
+
+
+def set_cpu_idle(state="all", disable=True, old_value=None):
+    """
+    Set/Reset cpu idle states for all cpus
+
+    :param state: cpuidle state number, default: all states
+    :param disable: whether to disable/enable given cpu idle state
+    :param old_value: whether to reset to old value if given
+    """
+    cpus_list = cpu_online_list()
+    if not old_value:
+        states = []
+        if 'all' in state:
+            states = range(1, len(glob.glob("/sys/devices/system/cpu/cpu0/cpuidle/state*")))
```

Comment:
Any reason to start with 1? It is possible to disable state0 so for testing purposes we should allow all, shouldn't we?

Reply:
physically not, but in `sys` it can (at least on my system). How about changing the name to `sleep` to emphasize it disables sleep-states and leaves the c0 state unchanged?

Reply:
I mean for testing purposes it might be useful to have `all` which disables really all states including the c0 (despite the fact whether it makes sense or not)